src/utils/validationUtil.ts

```diff
@@ -167,7 +167,7 @@ export class ValidationUtil {
       return missingMandatoryFields;
     }
     // odometerReading and odoMeterReadingUnits are required only for HGV and PSV
-    if (!odometerReading) {
+    if (!odometerReading && odometerReading !== 0) {
```

Comment:
An attempt was made to use JOI instead of the above if condition. The drawback for that is that the current version of JOI has limitation in customising the message. In order to use custom message with minimal change we have to upgrade JOI. It is suggested to use if condition for now and consider JOI upgrade as a separate ticket if the need arises.

Reply:
hmm the effort of upgrading JOI? we are on a really old version, aren't we?

Reply:
Yes we are on 14.x and the current release is 17.x

Reply:
For future reference, ticket created: https://jira.dvsacloud.uk/browse/CVSB-19278